src/server/server_family.cc

```diff
@@ -264,6 +358,44 @@ void ServerFamily::Load(const std::string& load_path) {
   });
 }
 
+void ServerFamily::SnapshotScheduling(const string &&time) {
+  auto timezone = GetFlag(FLAGS_save_schedule_use_utc) ? absl::UTCTimeZone() : absl::LocalTimeZone();
+  while (is_running_) {
+    std::unique_lock lk(snapshot_mu_);
+    snapshot_timer_cv_.wait_for(lk, std::chrono::seconds(20));
```

Comment:
you do not need snapshot_mu_, snapshot_timer_cv_ for this.

you can just use `this_fiber::sleep_for(20s)`.

Reply:
I had originally used that but on shutdown of a ServerFamily it could hang for up to 20s waiting for the snapshot fiber to wake up and join. With the cv I can wake up the thread before the 20s are up so the fiber joins (almost) instantly

Reply:
I just submitted a patch to helio: https://github.com/romange/helio/commit/32a5c82f5b4316e7f050dd90b8dfe5a34cee4ed0

that allows using WaitFor with done object. This object replaces the triple above. You can either stay with (mutex/cndvar/bool) triple or use the `Done` object, but then you will need to pull the latest version of helio into dragonfly. Whatever you choose is fine.

Reply:
I ended up using the `Done` object as it seemed cleaner, but needed a small change to make the `WaitFor` function public(https://github.com/romange/helio/pull/11)

Reply:
Merged